bin/faucet/src/api/get_tokens.rs

```diff
@@ -197,11 +194,16 @@ impl RawMintRequest {
         };
 
         let account_id = if self.account_id.starts_with("0x") {
-            AccountId::from_hex(&self.account_id)
+            AccountId::from_hex(&self.account_id).map_err(AccountError::ParseId)
         } else {
-            AccountId::from_bech32(&self.account_id).map(|(_, account_id)| account_id)
+            Address::from_bech32(&self.account_id)
+                .map_err(AccountError::ParseAddress)
+                .and_then(|(_, address)| match address {
+                    Address::AccountId(account_id_address) => Ok(account_id_address.id()),
+                    _ => Err(AccountError::AddressNotIdBased),
+                })
         }
-        .map_err(MintRequestError::AccountId)?;
+        .map_err(MintRequestError::AccountError)?;
```

Comment:
This is fine for now, but in the future we should make this more generic. Basically, based on address/id we should be able to figure out how to build notes. Specifics of this are still TBD and maybe a part of this should live in the client (or maybe in `miden-base`) - but let's create an issue to figure out how to build notes based on addresses.